qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py

```diff
@@ -176,14 +211,14 @@ def solve(
 
         return result
 
-    def _prepare_matrix_operators(self, problem) -> Tuple[dict, int]:
+    def _prepare_matrix_operators(self, problem: BaseProblem) -> Tuple[dict, int]:
         """Construct the excitation operators for each matrix element.
 
         Returns:
             a dictionary of all matrix elements operators and the number of excitations
             (or the size of the qEOM pseudo-eigenvalue problem)
         """
-        data = problem.hopping_qeom_ops(self._gsc.qubit_converter, self._excitations)
+        data = self._build_hopping_ops(problem)
         hopping_operators, type_of_commutativities, excitation_indices = data
```

Comment:
I'm actually wondering why the excitation_indices is needed. In the code, I think it is only used to compute the size in the following line which is simple to circumvent or to move around.
(It is  overall a useful piece of information, I'm just questioning its role here)

Reply:
Afaict it is indeed only used to compute `size` 2 lines below. `size` is then used to determine the EOM matrices size. Maybe this can be done in a more clever way.
In fact, this goes to show that #769 (which I opened shortly after this PR) is something that would be worthwhile looking into.

Reply:
Yes indeed, let's make it part of the refactoring